puppet/chassis.rb

```diff
@@ -229,4 +230,124 @@ def self.make_relative(base, paths)
 			pathname.relative_path_from(base).to_s
 		end
 	end
+
+	def self.update_core(config)
+		if config["auto_update"]["core"] == true
+			puts "\e[32mChecking for Chassis core updates...\e[0m"
+			updates = self.updates_check(['core'], @@dir)
+			if updates.empty?
+				puts "\033[A\e[K"
+			else
+				self.prompt_for_updates(updates, @@extension_dir, 'core')
+			end
+		end
+	end
+
+	def self.update_submodules(config)
+		if config["auto_update"]["submodules"] == true
+			puts "\033[A\e[32mChecking for Chassis submodule updates...\e[0m"
+			submodules = ['apt','mysql','stdlib','wp']
+			directory = File.join(@@dir, 'puppet/modules')
+			updates = self.submodule_update_check(submodules, directory)
+		end
+	end
+
+	def self.update_extensions(config)
+		if config["auto_update"]["extensions"] == true
+			puts "\033[A\e[32mChecking for Chassis extension updates...\e[0m"
+			extensions = Dir.glob(@@extension_dir + '/*').map { |directory| File.basename( directory ) }
+			updates = self.updates_check(extensions, @@extension_dir)
+			if updates.empty?
+				puts "\033[A\r\e[K"
+			else
+				self.prompt_for_updates(updates, @@extension_dir, 'extensions')
+			end
+		end
+	end
+
+	def self.update_global_extensions(config)
+		if config["auto_update"]["global_extensions"] == true
+			puts "\033[A\e[32mChecking for Chassis global extension updates...\e[0m"
+			global_ext_path = File.join(Dir.home, ".chassis", "extensions")
+			global_extensions = Dir.glob(global_ext_path + '/*').map { |directory| File.basename( directory ) }
+			updates = self.updates_check(global_extensions, global_ext_path)
+			if updates.empty?
+				puts "\033[A\r\e[K"
+			else
+				self.prompt_for_updates(updates, global_ext_path, 'extensions')
+			end
+		end
+	end
+
+	def self.updates_check(folders, directory)
+		updates = Array.new
+		folders.each do |folder|
+			next if folder == 'example'
+			if 'core' == folder
+				folder = directory
+				Dir.chdir(directory)
+			elsif
+				Dir.chdir(directory + '/' + folder )
+			end
+			_, _, _ = Open3.capture3("git checkout master")
+			_, _, _ = Open3.capture3("git remote update")
```

Comment:
Do we want to check the status return at all? If `checkout` fails e.g. do we want to continue the next commands or report an error and move to the next dir?

Reply:
I just tested if it errors by removing a `.git` folder in one of the extensions and it just continues which is good so I've not bother to return the statuses.